python/lsst/meas/astrom/match_probabilistic_task.py

```diff
@@ -71,17 +72,21 @@ def _apply_select_bool(
             The final selection array.
 
         """
+        is_pd = isinstance(catalog, pd.DataFrame)
         select_additional = (len(columns_true) + len(columns_false)) > 0
         if select_additional:
             if selection is None:
                 selection = np.ones(len(catalog), dtype=bool)
             for column in columns_true:
                 # This is intended for boolean columns, so the behaviour for non-boolean is not obvious
                 # More config options and/or using a ConfigurableActionField might be best
-                values = catalog[column].values
+                values = catalog[column]
+                if is_pd:
+                    values = values.values
```

Comment:
Pandas really forces into ugly constructions.  I don't love the look of this, though.  `values.values` is confusing.
Maybe something like
`catalog_values = catalog[column] if not is_pd else catalog[column].values`